spec/Appendix B -- Grammar Summary.md

```diff
@@ -47,7 +47,7 @@ Punctuator :: one of ! $ & ( ) ... : = @ [ ] { | }
 
 Name ::
 
-- NameStart NameContinue\* [lookahead != NameContinue]
+- NameStart NameContinue\* [lookahead != {NameContinue, `.`}]
```

Comment:
don't think we need this,

the reason for this quirk in the grammar is to remove ambiguity about greediness, e.g.

```
ThisIsAName
```

Where `ThisIs` would be a valid match for `NameStart NameContinue*`, but isn't what we want since we want it to greedily consume the rest of the following NameContinue, hence we say `[lookahead != NameContinue]` to make it clear that it's not valid to stop here and it must greedily continue before yielding a Name token